compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -265,14 +265,45 @@ public void visit(BLangPackage pkgNode) {
                 sortedListOfNodes.add(topLevelNode);
             }
         });
-        sortedListOfNodes.forEach(topLevelNode -> analyzeNode((BLangNode) topLevelNode, env));
+
+        for (TopLevelNode topLevelNode : sortedListOfNodes) {
```

Comment:
we need to analyze the init function before any other function.
Consider blow 
```ballerina
import ballerina/io;

public function main() {
    io:println("hello");
    io:println(s);
}

function __init() {
    s = "hello";
}
string s;
```

Reply:
Changed